testing/trino-testing/src/main/java/io/trino/testing/AbstractTestDynamicPartitionPruning.java

```diff
@@ -0,0 +1,390 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.testing;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.trino.Session;
+import io.trino.server.DynamicFilterService;
+import io.trino.spi.QueryId;
+import io.trino.spi.predicate.Domain;
+import io.trino.spi.predicate.ValueSet;
+import io.trino.tpch.TpchTable;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.airlift.testing.Assertions.assertGreaterThan;
+import static io.trino.SystemSessionProperties.ENABLE_LARGE_DYNAMIC_FILTERS;
+import static io.trino.SystemSessionProperties.JOIN_DISTRIBUTION_TYPE;
+import static io.trino.SystemSessionProperties.JOIN_REORDERING_STRATEGY;
+import static io.trino.spi.predicate.Domain.none;
+import static io.trino.spi.predicate.Domain.singleValue;
+import static io.trino.spi.predicate.Range.range;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.sql.analyzer.FeaturesConfig.JoinDistributionType.PARTITIONED;
+import static io.trino.sql.analyzer.FeaturesConfig.JoinReorderingStrategy.NONE;
+import static io.trino.tpch.TpchTable.LINE_ITEM;
+import static io.trino.tpch.TpchTable.ORDERS;
+import static io.trino.tpch.TpchTable.SUPPLIER;
+import static io.trino.util.DynamicFiltersTestUtil.getSimplifiedDomainString;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public abstract class AbstractTestDynamicPartitionPruning
```

Comment:
Currently this test is not helping us to verify correctness.
We should improve it to run every query with and without DF enabled and assert that results are same in both cases.

Reply:
Can you add extra commits so the rename `AbstractTestDynamicPartitionPruning` -> `TestHiveDynamicPartitionPruning` is actually reported as rename (with changes) instead of massive delete and massive add.

It should work if you add two intermediate commits:
1. Rename TestHiveDynamicPartitionPruning -> AbstractTestDynamicPartitionPruning, and add TestHiveDynamicPartitionPruningTemporary which extends AbstractTestDynamicPartitionPruning
2. Rename TestHiveDynamicPartitionPruningTemporary -> TestHiveDynamicPartitionPruning

Alternatively you can do just
1. Rename TestHiveDynamicPartitionPruning -> AbstractTestDynamicPartitionPruning
2. Add (re-add) TestHiveDynamicPartitionPruning which extends AbstractTestDynamicPartitionPruning

The latter will have one commit with worse test coverage, but I would not care about that too much.